### PR TITLE
HDFS Shell commands cheat sheet goodie

### PR DESCRIPTION
**Pull request by AshishAAB**
# HDFS Shell commands cheat sheet goodie

**What does your Instant Answer do?**
Lists the commonly used HDFS shell commands for interacting with files and directories over HDFS

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://hadoop.apache.org/docs/r2.7.1/hadoop-project-dist/hadoop-common/FileSystemShell.html#usage

**Why did you choose this data source?**
This is the official latest documentation of Apache Hadoop

**Are there any other alternative (better) data sources?**
Since this is the official documentation, there are no better sources.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
This will be mostly useful for DevOps working in the BigData field

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Yes (https://duck.co/ideas/idea/6228/cheat-sheet-for-hadoop-s-hdfs-shell-commands)

**Are you having any problems? Do you need our help with anything?**
Yes.
I was trying to split the commands into multiple lines so that I can give some alternatives of the command explained. There should be some provision for this. May be if we can specify multiple values(as a list) in the val section of the JSON which would then be rendered as multiple lines of text.

**Where did you hear about DuckDuckHack? (For first time contributors)**
Through a meetup group (mumbai meetup group)

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
image

image

-----
IA Page: https://duck.co/ia/view/hdfs_shell_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/hdfs-shell.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "description": "A brief overview of the commands for interacting with HDFS files and directories",
+  "metadata": {
+    "sourceName": "Apache Hadoop v2.7.1 documentation",
+    "sourceUrl": "http://hadoop.apache.org/docs/r2.7.1/hadoop-project-dist/hadoop-common/FileSystemShell.html"
+  },
+  "section_order": [
+    ""
+  ],
+  "sections": {
+    "": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/hdfs-shell.json]: Please provide a section name.

Is there any chance you could break this up into 2 or more sections? Any more content you could add?

File: share/goodie/cheat_sheets/hdfs-shell.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+  "id": "hdfs_shell_cheat_sheet",
+  "name": "HDFS Shell Commands",
+  "description": "A brief overview of the commands for interacting with HDFS files and directories",
+  "metadata": {
+    "sourceName": "Apache Hadoop v2.7.1 documentation",
+    "sourceUrl": "http://hadoop.apache.org/docs/r2.7.1/hadoop-project-dist/hadoop-common/FileSystemShell.html"
+  },
+  "section_order": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/hdfs-shell.json]: We *just* launched Cheat Sheet Templates. This one can use the new "terminal" template. Please add:

```json
"template_type": "terminal"
```

**Comment by moollaza** [comment]
@AshishAAB thanks a lot for contributing, we really appreciate it! Sorry for the delayed review! Somehow this one slipped by me.

**Comment by moollaza** [comment]
>There should be some provision for this. May be if we can specify multiple values(as a list) in the val section of the JSON which would then be rendered as multiple lines of text.

We may add this functionality later, but for now it's easiest to simply add another key-value pair, and indicate which ones are alternaitves by added `(alernative)` or similar in the description.

**Comment by moollaza** [comment]
@AshishAAB Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by moollaza** [comment]
I'm closing this one for now, but if you'd like to finish this up, we'll gladky open it back up for you!

**Comment by javathunderman** [comment]
@moollaza We can remove the "Abandoned" label seeing as #2067 has been merged.

**Pull request closed without merging**